Assets/Scripts/Entity/Player/PlayerState.cs

```diff
@@ -143,7 +144,8 @@ public void Hit(int damage)
         }
 
         public void Die()
-        {
+        {            
+            OnPlayerDeath?.Invoke(1);
```

Comment:
maybe we should also change line 93 from
`else if (updatedValue < 0)` to `else if (updatedValue <= 0)`
this mistake of mine hasn't been fixed yet, why not do it here?

Reply:
absolutely. will do